keyboards/preonic/keymaps/trigotometry/keymap.c

```diff
@@ -0,0 +1,204 @@
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |      | Prev | Play | Skip |      |      |   1  |   2  |   3  |   =  |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      | Supr |      |             |   0  | Home |PageDn|PageUp|  End |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = {
+  {KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   XXXXXXX, KC_PAST, KC_PSLS, XXXXXXX, KC_DEL },
+  {_______, XXXXXXX, KC_VOLD, KC_UP,   KC_VOLU, XXXXXXX, XXXXXXX, KC_7,    KC_8,    KC_9,    KC_PMNS, KC_BSPC},
+  {_______, XXXXXXX, KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX, KC_4,    KC_5,    KC_6,    KC_PPLS, XXXXXXX},
+  {_______, XXXXXXX, KC_MRWD, KC_MPLY, KC_MFFD, XXXXXXX, XXXXXXX, KC_1,    KC_2,    KC_3,    KC_PEQL, KC_ENT },
+  {_______, _______, _______, KC_LGUI, _______, _______, _______, KC_0,    KC_HOME, KC_PGDN, KC_PGUP, KC_END }
+},
+
+/* Raise
+ * ,-----------------------------------------------------------------------------------.
+ * |  F12 |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |  F7  |  F8  |  F9  |  F10 |  F11 |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |  \   |      |      |   /  |      |      |      |      |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |  |   |  [   |  {   |  (   |  <   |   >  |   )  |   }  |   ]  |   |  |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |      |      |      |  +   |  -   |   _  |   =  |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |Dvorak|Qwerty| Game | Reset|
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = {
+  {KC_F12,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11 },
+  {_______, XXXXXXX, XXXXXXX, XXXXXXX, KC_BSLS, XXXXXXX, XXXXXXX, KC_PSLS, XXXXXXX, XXXXXXX, XXXXXXX, _______},
+  {_______, KC_PIPE, KC_LBRC, KC_LCBR, KC_LPRN, KC_LABK, KC_RABK, KC_RPRN, KC_RCBR, KC_RBRC, KC_PIPE, XXXXXXX},
+  {_______, XXXXXXX, XXXXXXX, XXXXXXX, KC_PPLS, KC_PMNS, KC_UNDS, KC_EQL,  XXXXXXX, XXXXXXX, XXXXXXX, _______},
+  {_______, _______, _______, _______, _______, _______, _______, _______, DVORAK,  QWERTY,  GAMING,    RESET}
+}
+
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+
+        case DVORAK:
+          if (record->event.pressed) {
+            persistent_default_layer_set(1UL<<_DVORAK);
```

Comment:
Could you replace this (and the like) with the updated function? 

Eg `set_single_persistent_default_layer(_DVORAK)`